### ISSUE TITLE
Mime.getMimeType method throws an error when supplied $path argument imageInfo is not available

### ISSUE THREAD

**Issue by ejnshtein** [magento/magento2#32878]
### Preconditions (*)

1. OS - Arch Linux (Manjaro)
 PHP version - 7.4.16
 Composer version - 2.0.12
 Magento version - 2.4.2
2. Working on adding support Magento 2.4.2 in [create-magento-app](https://docs.create-magento-app.com/) I faced an issue when requesting a file it throws the following error in `Magento/Framework/Filesystem/Driver/File` in [Mime class getMimeType method](https://github.com/magento/magento2/blob/4898eb483f81570dc718e3add5a31f0fa90599c3/lib/internal/Magento/Framework/Filesystem/Driver/File/Mime.php#L115).
This happens because the `getimagesize` function (on line 115) can return either `array` or `false` and there's no check if this method returned actually `array`.
![Screenshot_20210426_174623](https://user-images.githubusercontent.com/18352350/116102369-55953580-a6b7-11eb-94f1-0a362bcf3584.png)
So it dies with the following error in each request:
![Screenshot_20210426_173022](https://user-images.githubusercontent.com/18352350/116101086-35b14200-a6b6-11eb-91cc-e32c43560d3c.png)

### Steps to reproduce (*)

1. Install the alpha version of CMA.
```bash
# create project 
npx create-magento-app magento-app

# go inside
cd magento-app

# install an alpha version
npm i @scandipwa/magento-scripts@alpha

# start the installation and during setup choose magento 2.4.2 version.
npm start
```
2. Open Magento Admin panel on suggested URL and see errors in devtools network section.
![Screenshot_20210426_192138](https://user-images.githubusercontent.com/18352350/116117224-a95a4b80-a6c4-11eb-87ca-b86606969c62.png)
And if we go to one of the URLs we will see the following error.
![Screenshot_20210426_173022](https://user-images.githubusercontent.com/18352350/116101086-35b14200-a6b6-11eb-91cc-e32c43560d3c.png)

### Expected result (*)

1. If we add a simple `if` statement to `Magento/Framework/Filesystem/Driver/File` in [Mime class getMimeType method](https://github.com/magento/magento2/blob/4898eb483f81570dc718e3add5a31f0fa90599c3/lib/internal/Magento/Framework/Filesystem/Driver/File/Mime.php#L115) it works exactly as expected.
![Screenshot_20210426_174722](https://user-images.githubusercontent.com/18352350/116102532-765d8b00-a6b7-11eb-9258-46917ee93d0c.png)
![Screenshot_20210426_174813](https://user-images.githubusercontent.com/18352350/116102679-94c38680-a6b7-11eb-826e-1acced567e86.png)
![Screenshot_20210426_192225](https://user-images.githubusercontent.com/18352350/116117308-bf680c00-a6c4-11eb-923c-16cf0b344b45.png)

### Actual result (*)

1. ![Screenshot_20210426_192138](https://user-images.githubusercontent.com/18352350/116117224-a95a4b80-a6c4-11eb-87ca-b86606969c62.png)
2. ![Screenshot_20210426_173022](https://user-images.githubusercontent.com/18352350/116101086-35b14200-a6b6-11eb-91cc-e32c43560d3c.png)

This problem prevented static files to be loaded in a luma theme and admin panel.

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by mrtuvn** [comment]
Can you create PR for this ?

**Comment by ejnshtein** [comment]
Okay, I'll create a PR, but not sure about tests, never wrote them for a PHP project, so I'll try my best.
The whole point of CMA is to create an isolated environment to run Magento in, so it should behave the same way if you do Magento env setup on your system.

Regarding the issue investigation, it looks like it doesn't like when the file inside the `pub/static` folder is a symlink.
Example:
I have requests to 2 files in `pub/static/adminhtml/Magento/backend/en_US` folder:
http://localhost/static/version1619441477/adminhtml/Magento/backend/en_US/requirejs-config.js and
http://localhost/static/version1619441477/adminhtml/Magento/backend/en_US/jquery.js

`requirejs-config.js` is an actual file that is located in this folder, but `jquery.js` is a symlinked file and so I don't have this error for the `requirejs-config.js` file but have it for `jquery.js`:
![Screenshot_20210427_140009](https://user-images.githubusercontent.com/18352350/116231008-0277bd00-a761-11eb-9594-3d1fafef0836.png)
![Screenshot_20210427_140128](https://user-images.githubusercontent.com/18352350/116231065-13c0c980-a761-11eb-9133-bdd59afdeadf.png)
![Screenshot_20210427_140157](https://user-images.githubusercontent.com/18352350/116231110-23401280-a761-11eb-90b5-971c216e8f0f.png)

On the surface it looks like this to me, what do you think?

**Comment by Den4ik** [comment]
Hi @ejnshtein 

Based on PHP documentation FileInfo extension bundled and enabled by default. https://www.php.net/manual/en/fileinfo.installation.php

I believe that use of native functional is better and we should add `ext-fileinfo` as required extension instead of using workarounds. I think this question should be discussed with core team.
@gabrieldagama could you assist with it?

**Comment by ejnshtein** [comment]
Well, this is not exactly a workaround, since it's expected that `getimagesize` function returns array or false, it's just currently this code does not handle this behavior since it expect always an array.

**Comment by ejnshtein** [comment]
Tried with the `fileinfo` extension installed and it works as expected.
Nice.
For me, that means that I can just modify the config for Magento 2.4.2 in CMA so it will work now, probably will release a new version of CMA this week with the following changes.
As for magento source code, I don't think that the PR I send is unnecessary and would recommend merging it to the main branch, just to avoid possible issues, like the one that I reported here.

Thanks for the help!
